### PR TITLE
feat: add support for `user-event`

### LINKED ISSUE #10: support `testing-library/user-event`
I haven't worked out the details, but enabling users to play with  [user-event]s would awesome. Perhaps together with adding support for [HTML mixed mode]?

[HTML mixed mode]: https://codemirror.net/mode/htmlmixed/index.html
[user-event]: https://testing-library.com/docs/ecosystem-user-event

---

inspiration: https://twitter.com/kentcdodds/status/1270919381380284416?s=20

### LINKED ISSUE #69: sandbox markup
Just like #68, we need to create a sandbox for the markup.

Currently, it's possible to add a `<meta http-equiv="refresh" content="1; URL=http://evil.corp">` to the HTML markup, and trigger a full page refresh.

This shouldn't be possible.

We should render the markup in an `iframe` with the proper restrictions. (See #68). It's quite easy to render the markup (with the `srcdoc` attribute). But we also need to inject the custom scrollbars, and tailwind styling. That's what will consume the most time of this ticket.

### PR DESCRIPTION
**Pull request by smeijer**
### What:

I've done a number of things in the PR. 

#### Sandboxed the preview pane

  - The preview is now rendering in an iframe
    Communication with the top frame is handled through the `postMessage` API.

#### Highlighting

  - The iframe has been made responsible for highlighting elements, which now uses the same Highlighter as the chrome devtool

#### Parser

  - `@testing-library/user-events` has been added to the evaluator context. User events are available under `user` and `userEvent`.

#### Query Pane

  - The query pane now has two menu buttons:

    - `run`
      The run button enables the user to execute queries manually. It's also possible to trigger this action with `cmd + enter` or `ctrl + enter`.     While a query is being evaluated, a `loader` is rendered instead.

    - `auto sync`
      This button enables/disables automatic synchronization. It also allows the user to toggle between the pre/post query state in the preview pane. Some users might find the automatic evaluation confusing when working with user-events. This button might offer some help.

#### Result Pane
  
  - now renders the same image as the `Preview` pane when there is no query

#### Generic

   - icons are now loaded from `@primer/octicons-react`
   
#### Devtools

   - The devtools are updated to NOT show the `run` and `sync` button.

### Why:
Because it's awesome!

### How:

Check the code and "What" above. It's a lot!

**Checklist**:

- [ ] Tests
- [x] Ready to be merged
      

---

resolves #10 
resolves #69 

**todo**
- [x] toggle between pre and post markup
- [x] add some `parsing` indicators
- [x] evaluate on paste
- [x] fix the suggestion pane
- [ ] move new navigation bar higher up the tree
- [ ] cleanup

testing-playground-user-events

### CODE REVIEW

File: src/sandbox.js [hunk 1]
```diff
@@ -1,71 +1,152 @@
...
+  state.queriedNodes = result.elements.map((elem) => elem.target);
+  state.highlighter.highlight({ nodes: state.queriedNodes });
+}
+
+function setInnerHTML(node, html) {
+  const doc = node.ownerDocument;
+  node.innerHTML = html;
+
+  for (let prevScript of node.querySelectorAll('script')) {
+    const newScript = doc.createElement('script');
```
Comment (delca85) [inline src/sandbox.js:40]: In this way, aren't you getting (and replacing with js functions) all the `script` elements, not only the `javascript` ones?

Comment (smeijer) [reply src/sandbox.js:40]: Thanks :+1: . Fixed

**Comment by smeijer** [comment]
ezgif com-video-to-gif (1)

**Comment by smeijer** [comment]
Ok, I don't have much time this week. So I'm going to merge this tomorrow unless there are blocking issues that I've missed.

I still want the menu in a different position in the tree, and I should have cleaned up the parser. But that can be done in a new PR once there is time.

If anyone is reading this before I have pushed the merge button, and feeling to test/review, please take a shot. It would be much appreciated.

**Comment by delca85** [comment]
I have noticed this problem:
sometime, when I enter a query or when I click an element on preview to insert the query to get it, the new code go above the old one and the result is the one you found in the attached image.

I am using Chrome in a Mac Os environment.
Screenshot 2020-06-23 at 21 43 24

**Comment by smeijer** [comment]
@delca85 , it looks like there are two CodeMirror instances bound to the same element.

Do you remember which steps you took to trigger that problem? A reproduction would be very helpful.

**Comment by delca85** [comment]
I am not able to reproduce this behavior right now.
Yesterday, what I have done was:
1. open testing playground with the default content
2. adding and running a query below the comments inserted by default in `QueryEditor`
3. click on an element in `Preview`
4. the query obtained by click an element in step 3 appears above the comments in `QueryEditor`

**Pull request merged**